pyproject_metadata/project_table.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+import sys
+import typing
+
+
+if typing.TYPE_CHECKING:
+    from typing import Any, Mapping, Sequence, Union
+
+    if sys.version_info < (3, 11):
+        from typing_extensions import Required
+    else:
+        from typing import Required
+
+    if sys.version_info < (3, 8):
+        from typing_extensions import TypedDict
+    else:
+        from typing import TypedDict
+
+
+__all__ = [
+    'ContactTable',
+    'ReadmeTable',
+    'ProjectTable',
+    'BuildSystemTable',
+    'PyProjectTable',
+]
+
+
+def __dir__() -> list[str]:
+    return __all__
+
+
+class ContactTable(TypedDict, total=False):
+    name: str
+    email: str
+
+
+ReadmeTable = TypedDict(
```

Comment:
I see some uses class annotation why others use `TypeDict` function. Any concern not keeping them consistent?

Reply:
I used class annotation where possible, but you can't add a `-` without using the less powerful function form.

Reply:
(And Ruff rewrites functional form to class form if possible, so I'd have to manually ignore it to get the one or so class form without `-` into functional form.)

Reply:
Yes, I just noticed.